src/libsystemd/sd-device/sd-device.c

```diff
@@ -1171,24 +1171,11 @@ _public_ int sd_device_get_devname(sd_device *device, const char **devname) {
 static int device_set_sysname_and_sysnum(sd_device *device) {
         _cleanup_free_ char *sysname = NULL;
         const char *sysnum = NULL;
-        const char *pos;
         size_t len = 0;
 
-        if (!device->devpath)
```

Comment:
did you intend to remove this check? if so, it should have an assert

Reply:
The condition should be already handled by `path_extract_filename()`.

Reply:
but `device` is dereferenced before that runs?

Reply:
Ah, indeed. Will fix it. Thanks.